### PR TITLE
Updated to use secp256k1 tags. Using: v0.3.2

### PR DESCRIPTION
**Pull request by MementoRC**
Proposing to update to the latest version of `secp256k1` release using its tags. Current version is `v0.3.2`

This is a simple drop-in that passes the testsuite. However, there are warnings of deprecated functions usage. My understanding of this package is superficial. The project that I work on uses `coincurve` and `conda`, I am motivated to get a `conda-recipe` to upgrade the conda dependencies of the project

### CODE REVIEW

**Review by ofek — CHANGES_REQUESTED** [review]
Can you please continue using a commit hash for reproducibility? Also, you'll need to update this file https://github.com/ofek/coincurve/blob/master/coincurve/_windows_libsecp256k1.py

**Comment by MementoRC** [comment]
@ofek There are a few changes that are not strictly necessary (change in actions revisions, macos version, etc ...) and a few that I don't quite understand (change from -0.dll to -2.dll?). Let me know your thoughts and if changes are needed

I am also looking at additional PRs:
    - Update of setup.py to allow building the conda-forge package
        Currently it fails to build on Windows
        Linux/MacOS builds but with patches in the recipe
    - Add support for using the CMake that `secp256k1` seems to be implementing (still experimental)

**Comment by MementoRC** [comment]
@ofek , just FYI, I finally got the conda-recipe to partially pass. I battled for 2 days with the win-VC/python-3.8 build, but it fails on for some odd reasons (not coincurve). The PR for the recipe is under:
https://github.com/conda-forge/staged-recipes/pull/23741
I apply a few patches (some turn out to be unnecessary), notably to the `setup.py` that I submitted in another PR for your review

**Comment by ofek** [comment]
Nice job! What would you like me to do/merge first?

**Comment by MementoRC** [comment]
@ofek, do you disapprove this PR?

File: setup.py [hunk 1]
```diff
@@ -35,11 +35,12 @@
 # IMPORTANT: keep in sync with .github/workflows/build.yml
 #
 # Version of libsecp256k1 to download if none exists in the `libsecp256k1` directory
-UPSTREAM_REF = os.getenv('COINCURVE_UPSTREAM_REF') or 'ddf2b2910eb19032f8dd657c66735115ae24bfba'
+# UPSTREAM_TAG = os.getenv('COINCURVE_UPSTREAM_TAG') or 'v0.3.2'
```
Comment (ofek) [inline setup.py:38]:
```suggestion
```

File: setup.py [hunk 2]
```diff
@@ -35,11 +35,12 @@
...
+UPSTREAM_REF = os.getenv('COINCURVE_UPSTREAM_TAG') or 'acf5c55ae6a94e5ca847e07def40427547876101'
 
+# LIB_TARBALL_URL = f'https://github.com/bitcoin-core/secp256k1/archive/refs/tags/{UPSTREAM_TAG}.tar.gz'
```
Comment (ofek) [inline setup.py:41]:
```suggestion
```

File: setup_support.py [hunk 3]
```diff
@@ -68,24 +64,15 @@ def _find_lib():
     ffi = FFI()
     try:
         ffi.dlopen('secp256k1')
-        if os.path.exists('/usr/include/secp256k1_ecdh.h'):
-            return True
-        else:
-            # The system library lacks the ecdh module
-            return False
+        return bool(os.path.exists('/usr/include/secp256k1_ecdh.h'))
```
Comment (ofek) [inline setup_support.py:67]:
```suggestion
        return os.path.exists('/usr/include/secp256k1_ecdh.h')
```

**Review by ofek — CHANGES_REQUESTED** [review]
I'm a bit behind on open source, sorry about the late reply!

File: setup.py (hunk 1 above)
Comment (MementoRC) [reply setup.py:38]: For my understanding, why is the REF preferred to the TAG?

Comment (ofek) [reply setup.py]: tags can be altered to point to different commits but commit hashes can never change

**Comment by MementoRC** [comment]
> I'm a bit behind on open source, sorry about the late reply!

Sure, I hesitated to pester ;P  For the conda recipe, I am thinking about creating a libsecp256k1 recipe and will modify the coincurve recipe to dependent on that conda package, this way, coincurve conda recipe will not need to build the library (still experimenting though)

By the way, do you want to be listed as a maintainer for the conda recipe?

**Review by ofek — APPROVED** [review]

**Comment by ofek** [comment]
> do you want to be listed as a maintainer for the conda recipe?

Sure!

**Pull request merged**